src/ring_theory/fractional_ideal.lean

```diff
@@ -651,10 +651,11 @@ begin
     exact hy' }
 end
 
-instance is_principal {R} [principal_ideal_domain R] {f : fraction_map R K}
+instance is_principal {R} [integral_domain R] [is_principal_ideal_ring R] {f : fraction_map R K}
   (I : fractional_ideal f) : (I : submodule R f.codomain).is_principal :=
 ⟨ begin
   obtain ⟨a, aI, ha⟩ := exists_eq_span_singleton_mul I,
+  haveI : is_principal aI := is_principal_ideal_ring.principal aI,
```

Comment:
You deleted the line making `is_principal_ideal_ring.principal` an instance in `principal_ideal_domain.lean`. If you add this back do you still need this line?

Reply:
I think that was an overzealous deletion; I put it back and removed a couple more haveI's.